src/types.rs

```diff
@@ -1653,46 +2190,222 @@ impl TRS {
-    /// The number of rules in the TRS.
+    /// The number of [`Rule`]s in the `TRS`.
+    /// 
+    /// [`Rule`]: struct.Rule.html
+    ///
+    /// # Examples
+    /// ```
+    /// # use term_rewriting::{Signature, TRS, parse_trs};
+    /// let mut sig = Signature::default();
+    ///
+    /// let t = parse_trs(&mut sig,
+    /// "A = B;
+    /// C = D | E;
+    /// F(x_) = G;").expect("parse of A = B; C = D | E; F(x_) = G;");
+    /// 
+    /// assert_eq!(t.len(), 3);
+    /// ```
     pub fn len(&self) -> usize {
         self.rules.len()
     }
-    /// Are there any rules in the TRS?
+    /// Are there any [`Rule`]s in the `TRS`?
+    /// 
+    /// [`Rule`]: struct.Rule.html
+    ///
+    /// # Examples
+    /// ```
+    /// # use term_rewriting::{Signature, TRS, parse_trs};
+    /// let mut sig = Signature::default();
+    ///
+    /// let t = parse_trs(&mut sig,
+    /// "A = B;
+    /// C = D | E;
+    /// F(x_) = G;").expect("parse of A = B; C = D | E; F(x_) = G;");
+    /// 
+    /// assert!(!t.is_empty());
+    ///
+    /// let t = parse_trs(&mut sig, "").expect("parse of blank string");
+    /// 
+    /// assert!(t.is_empty());
+    /// ```
     pub fn is_empty(&self) -> bool {
         self.rules.is_empty()
     }
-    /// Return the number of total number of subterms across all rules.
+    /// Return the number of total number of subterms across all [`Rule`]s in the `TRS`.
+    ///
+    /// See [`Term`] for more information.
+    ///
+    /// [`Term`]: struct.Term.html
+    /// [`Rule`]: struct.Rule.html
+    /// 
+    /// # Examples
+    ///
+    /// ```
+    /// # use term_rewriting::{Signature, TRS, parse_trs};
+    /// let mut sig = Signature::default();
+    ///
+    /// let t = parse_trs(&mut sig,
+    /// "A = B;
+    /// C = D | E;
+    /// F(x_) = G;").expect("parse of A = B; C = D | E; F(x_) = G;");
+    /// 
+    /// assert_eq!(t.size(), 8);
+    /// 
+    /// let t = parse_trs(&mut sig, 
+    /// "A(x_) = B;
+    /// C = D;").expect("parse of A(x_) = B; C = D;");
+    ///
+    /// assert_eq!(t.size(), 5);
+    /// ```
     pub fn size(&self) -> usize {
         self.rules.iter().map(Rule::size).sum()
     }
-    /// A serialized representation of the TRS.
+    /// A serialized representation of the `TRS`.
+    /// 
+    /// # Examples
+    ///
+    /// ```
+    /// # use term_rewriting::{Signature, TRS, parse_trs};
+    /// let mut sig = Signature::default();
+    ///
+    /// let t = parse_trs(&mut sig,
+    /// "A = B;
+    /// C = D | E;
+    /// F(x_) = G;").expect("parse of A = B; C = D | E; F(x_) = G;");
+    /// 
+    /// assert_eq!(t.display(&sig), 
+    /// "A = B;
+    /// C = D | E;
+    /// F(x_) = G;");
+    /// ```
     pub fn display(&self, sig: &Signature) -> String {
         self.rules
             .iter()
             .map(|r| format!("{};", r.display(sig)))
             .join("\n")
     }
-    /// A human-readable serialization of the TRS.
+    /// A human-readable serialization of the `TRS`.
```

Comment:
Can you please add an example highlighting the difference between `pretty` and `display`?

Reply:
Disccussed in person - will be part of a separate pull request